dev/golangci/dagger.json

```diff
@@ -0,0 +1,6 @@
+{
+  "name": "golangci",
+  "sdk": "go",
+  "source": ".",
+  "engineVersion": "dev-bb9c713a37c73cd4176da27c1e0c4654682f4de7"
```

Comment:
These versions are now significant, the API this wants is v0.11.9.

Reply:
My bad, I will correct this.

- Question: how are dev versions interpret by compat mode?
- Observation: this makes it even more annoying that switching back and forth between dev and stable engines on the same machine causes the engine container to be wiped

Reply:
There's some improvements to this in https://github.com/dagger/dagger/pull/7858

However, compat mode and dev versions are very fiddly. The current behavior (and in that pr) is that a dev version is *always* considered to consume the latest API and schema.

I'm actively thinking about how to get this to work in a reasonable way - it's very much a work in progress.

Reply:
> However, compat mode and dev versions are very fiddly. The current behavior (and in that pr) is that a dev version is always considered to consume the latest API and schema.

That seems reasonable! See [this discord thread](https://discord.com/channels/707636530424053791/1260743246917795840) about the need for some sort of flag to control compat mode explicitly.